src/coreclr/jit/lsra.cpp

```diff
@@ -8312,6 +8320,12 @@ void LinearScan::handleOutgoingCriticalEdges(BasicBlock* block)
             GenTree* op1 = lastNode->gtGetOp1();
             switchRegs |= genRegMask(op1->GetRegNum());
 
+            if (op1->OperIs(GT_COPY))
```

Comment:
Are locals and copies the only special cases that might happen here?

Reply:
Also `switch_regs` is not as descriptive as one might like, given that we now set it for things that are not switches.

Reply:
That's the only ones I could think of. `COPY` are special because we defer the code generation until the consuming node like here:

https://github.com/dotnet/runtime/blob/cb606ad46457dd2e9e11f1d9b954f239b2d9a98a/src/coreclr/jit/codegenarmarch.cpp#L379-L381

So in the test that I was investigating, the `GT_COPY` node's generation get deferred and happens after the generation of resolution move, at the time when the `JCMP` is generated.

```
Generating: N787 ( 34, 17) [000046] --CXGO------        t46 = *  CALLV vt-ind int    MiniBench.MyChannelWriter`1[Int32][System.Int32].TryWrite REG x0 $218
							GC regs: 100001 {x0 x20} => 100000 {x20}
Call: GCvars=0000000000000008 {V00}, gcrefRegs=100000 {x20}, byrefRegs=80000 {x19}
[15] Rec call GC vars = 0000000000000008
IN000f:                           blr     x21
                                                              /--*  t46    int    
Generating:                [000930] --CXGO------       t930 = *  COPY      int    REG x21 (currently: ***not unused***, should be unused)
Generating: N789 (  1,  2) [000047] -c----------        t47 =    CNS_INT   int    0 REG NA $40
; Generating: N001 (  1,  1) [000937] ------------       t937 =    LCL_VAR   byref  V00 this          x19 REG x19
                                                              /--*  t937   byref  
Generating: N002 (  2,  2) [000938] ------------       t938 = *  COPY      byref  REG x0 (currently: ***unused***, should be notunused)
IN0010:                           mov     x0, x19
							V00 in reg x19 is becoming dead  [000937]
							Live regs: 180000 {x19 x20} => 100000 {x20}
							Byref regs: 80000 {x19} => 0000 {}
							V00 in reg x0 is becoming live  [000938]
							Live regs: 100000 {x20} => 100001 {x0 x20}
							Byref regs: 0000 {} => 0001 {x0}
                                                              /--*  t930   int    
                                                              +--*  t47    int    
Generating: N791 ( 36, 20) [000048] CNE---XGO-N----              *  JCMP      void   REG NA
IN0011:                           mov     w21, w0
IN0012:                           cbnz    (LARGEJMP)L_M2478_BB37
```

Regarding `switchRegs`, I will rename it to `consumedRegs`.